src/nvme/fabrics.c

```diff
@@ -357,10 +357,18 @@ static int __add_argument(char **argstr, const char *tok, const char *arg)
 	return 0;
 }
 
+static int __nvmf_supported_options(nvme_root_t r);
+#define nvmf_check_option(r, tok)					\
+({									\
+	if (!(r)->options)						\
+		__nvmf_supported_options(r);				\
+	(r)->options->tok;						\
```

Comment:
If `__nvmf_supported_options()` fails to allocate memory, it can return with `(r)->options` still NULL, which would lead to a segfault. Should we check the return value?

Reply:
Yeah makes sense, though IIRC such allocations will never fail, instead the program gets terminated by the OOM killer.